pandas/tests/io/test_pytables.py

```diff
@@ -4927,6 +4927,23 @@ def test_categorical_conversion(self):
             result = read_hdf(path, 'df', where='obsids=B')
             tm.assert_frame_equal(result, expected)
 
+    def test_categorical_nan_only_columns(self):
+        # GH18413
+        # Check that read_hdf with categorical columns with NaN-only values can
+        # be read back.
+        df = pd.DataFrame({
+            'a': ['a', 'b', 'c', np.nan],
+            'b': [np.nan, np.nan, np.nan, np.nan],
+            'c': [1, 2, 3, 4]
+        })
+        df['a'] = df.a.astype('category')
+        df['b'] = df.b.astype('category')
+        expected = df.copy()
```

Comment:
you don't need to copy here

Reply:
true

Reply:
v0.22.0.txt  or 0.21.1? - I added it to 0.22.0 as per @jschendel's request.

Reply:
Go with what @jreback said, he'd know better than me.  I was just basing 0.22.0 off the issue originally being tagged as "Next Major Release", but that can change (which it has).

Reply:
done